docs/pages/machine-id/guides/github-actions-kubernetes.mdx

```diff
@@ -0,0 +1,221 @@
+metadata:
+  name: github-demo-kube-access
+version: v5
+spec:
+  allow:
+    kubernetes_labels:
+      # This grants access to any Kubernetes cluster attached to your Teleport
+      # cluster.
+      '*': '*'
+    kubernetes_groups:
+      # This states that the `github-demo` group will be applied to your bot's
+      # requests to your Kubernetes cluster. In a later step, we will bind this
+      # group to a Kubernetes role with a ClusterRoleBinding.
+      - github-demo
+  deny: {}
+```
+
+Apply this to your Teleport cluster:
+
+```code
+$ tctl create -f role.yaml
+```
+
+Now create the Machine ID bot, assigning it the role you created above:
+
+```code
+$ tctl bots add github-demo --roles=github-demo-kube-access --token=github-token
+```
+
+## Step 3/4. Configure a Kubernetes Cluster Role Binding
+
+With the bot configured and a Teleport role that will pass a group to
+Kubernetes with the bot's requests, you now need a Role Binding to associate a
+Kubernetes role to that group.
+
+This is the stage where you will actually determine what level of access the
+bot will have when accessing your cluster. In our example, we are applying the
+`view` ClusterRole which will allow our bot to read, but not write, all
+resources in the Kubernetes cluster.
+
+In your case you may wish to create a custom Kubernetes role that lists the
+exact permissions that you want your bot to have. You may also wish to use a
+RoleBinding rather than a ClusterRoleBinding to limit the bot's access to an
+individual Kubernetes namespace.
+
+Create a file named `clusterolebinding.yaml` and insert:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: github-demo-group
+subjects:
+  - kind: Group
+    # Note that this name field matches the group name we provided in the
+    # Teleport role configuration.
+    name: github-demo
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  # "view" is a default ClusterRole that grants read-only access to resources
+  # See: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles
+  name: view
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Apply this to your cluster using `kubectl`:
+
+```
+$ kubectl apply -f ./clusterolebinding.yaml
+```
+
+## Step 4/4. Create the GitHub Actions workflow
+
+With all the configuration in Teleport and Kubernetes complete, you can now
+create a GitHub Actions workflow that can connect to your Kubernetes cluster.
+
+Our example workflow will list all of the pods contained within the cluster,
+but this could equally be modified to deploy to a Kubernetes cluster with
+`kubectl` or `helm`.
+
+In the GitHub workflows directory of your repository (`.github/workflows/`)
+create a new workflow YAML file, in this case `actionstest.yml`.
+
+Insert the following, ensuring that you replace `my-kubernetes-cluster` with the
+name of your Kubernetes cluster and `example.domain:443` with the address of
+your Teleport Proxy.
```

Comment:
To be more accommodating to cloud users, you can append "or cloud tenant (e.g. `example.teleport.sh`)".